### DISCUSSION TITLE
Using Brightfield Toluidine Blue images for analysis

### DISCUSSION THREAD

**Discussion by gbendale** [neuropoly/axondeepseg#639] [Help needed]
Hi, I downloaded and installed axondeepseg and was able to open up FSLEYES interface. After I try to open the tiff file, it prompts me to enter pixel info. For my images my pixel ratio is 0.1. After I click enter, it does not do anything past that. I'm not sure what I am missing. I followed the steps in the tutorial on the ADS web page. I am new at using python. Maybe I'm missing some important step?

Thanks,
G

**Comment by hermancollin** [comment]
Hi @gbendale thank you for reaching out. Could you answer these questions please:

1. What operating system are you on?
2.  The terminal you used to open `fsleyes` will also contain the ADS "debug" and logging output. Could you share the content of this console with us? Otherwise, it will be hard for us to assist you.

My guess is that either an error occured or the model was misused so the computation is unreasonably long.

**Reply by gbendale** [reply]
Hi @hermancollin 
I am on a Windows computer.  See belo for the debug op. It sasy that ads was correctly installed. Is this what you are looking for?

![image](https://user-images.githubusercontent.com/105873059/169345410-042c08a5-b974-4767-948b-1e7f389a9c8f.png)

**Reply by hermancollin** [reply]
Thank you. At least, the quick tests all pass, which is a good start.

However, this is not the output I am looking for. How did you open fsleyes? In our normal workflow, we enter the `fsleyes` command in a terminal, which opens the app. This is where the ADS output will be printed when you use the ADS plugin in fsleyes. On your capture, I can see you did not type the fsleyes command. Maybe you did in a separate anaconda prompt?

**Comment by gbendale** [comment]
Oh I see. This is what I have.

![image](https://user-images.githubusercontent.com/105873059/169349956-b040aa39-4b6a-4a4b-97d5-f16134bff238.png)

**Reply by hermancollin** [reply]
Ah this is the FSLeyes window. Apologies if my instructions are unclear, but this is still not what I mean.
Here is a video tutorial on the ADS install process: https://youtu.be/7RwZihR0HWk?t=360
At 6 min into the video, we type `fsleyes` to open the plugin. The terminal window in which we type this command is what I am looking for here.

**Reply by gbendale** [reply]
Sorry, Im new to this so don't really know all the key words. This is where I type in fsleyes to open the window.

![image](https://user-images.githubusercontent.com/105873059/169353032-1a1893c5-bf1d-4c36-9311-f431a75d25b9.png)

**Reply by hermancollin** [reply]
@gbendale no problem we want to be accessible to everyone.

This is indeed the right console. It is weird that no further output is shown there after you typed the command, because when ADS segments an image, it outputs some debug information from the model being applied on the image. 

Could you describe the steps you take and what happens? For example, when you use the `Load PNG or TIF file` button, is the image correctly loaded on the screen? I understand you entered the pixel size, which tells me you probably clicked on the `Apply ADS prediction model` button afterwards. Was any error shown on screen? If the software correctly works, when you apply the prediction model, the window you just showed me (with the `fsleyes` command that you typed) should provide further information.

**Comment by gbendale** [comment]
Sure. Below are the steps I take and the op in the terminal and the plugin
The image loads correctly
![image](https://user-images.githubusercontent.com/105873059/169356298-6dac3e34-3d00-4970-bca9-f476f9b84f82.png)
  and the terminal shows this op
![image](https://user-images.githubusercontent.com/105873059/169356377-9a8958b8-3eb4-4797-9b6e-f6fd1f442e36.png)

then I apply the bf model and click apply ADS model
![image](https://user-images.githubusercontent.com/105873059/169356588-d409fa52-f83c-4cd5-8c01-e71b97231c0d.png)

then I input the pixel ratio as 0.1 and then nothing happens in the plug in or the terminal

**Reply by hermancollin** [reply]
Thank you for this detailed answer. This is indeed a strange behavior. 

I would like to try something. Would you be able to change directory in your anaconda prompt using `cd` to go inside the folder containing your image? Then, run the following command:
```
axondeepseg -t BF -i <NAME_OF_YOUR_IMAGE> -s 0.1
```
This should provide more information. Tell me if you encounter any problem.

**Reply by gbendale** [reply]
Still doesnt work. this is what I see. What am I doing wrong?
![image](https://user-images.githubusercontent.com/105873059/169373186-2304f267-9d78-4089-83cf-62a68738321a.png)

**Reply by hermancollin** [reply]
Ah yes sorry about that. Simply remove the `<` and `>` around the filename, like so:
```
axondeepseg -t BF -i 2_40x_ImageMid -s 0.1
```

**Comment by hermancollin** [comment]
I realize this back and forth could take a while. If you prefer, we could setup a zoom call or something similar to debug your problem so that we can share our screens.

**Reply by mathieuboudreau** [reply]
Yes @gbendale, seems like scheduling a zoom call with @hermancollin and/or our team of developers could speed up the solution to your problem. Please reach out to us if you are able and willing!

**Reply by gbendale** [reply]
Yes, I tried what you suggested and it still isnt working. I agree, a zoom call will be helpful. I am available next week in the afternoons to set one up. Who should I email?

**Comment by hermancollin** [comment]
@gbendale you can email the group at [axondeepseg@googlegroups.com](mailto:axondeepseg@googlegroups.com) and I'll answer you asap to setup the call.

**Reply by hermancollin** [reply]
(or directly email me at armand.collin@polymtl.ca if the above email fails)
